cuda/base/types.hpp

```diff
@@ -150,12 +150,26 @@ using cuda_type = typename detail::cuda_type_impl<T>::type;
  * @return `val` reinterpreted to CUDA type
```

Comment:
Just a comment: strictly speaking, for the second function it's not `reinterpreted` anymore but if we do not interpret this as the C++ `reinterpret` it makes sense anyway.

Reply:
It's a good point.

What `as_cuda_type` should be doing is converting between types that are "equivalent", but may have slightly different implementation on CUDA and on host code. However, they need to be ABI-compatible, so reinterpret cast of the same memory should work correctly.

So, the documentation is correct. What may not be correct is the implementation. While most such types should have a conversion operator, so `static_cast` should work correctly, that is not the requirement.

It may be better to implement the new version as `return *reinterpret_cast<cuda_type<T>*>(&val);`

Reply:
The new version seems good!